### DISCUSSION TITLE
[Question] Is there any way to keep division unchanged instead of Mul(x, Pow(y, -1))？

### DISCUSSION THREAD

**Discussion by HydrogenSulfate** [sympy/sympy#25668] [General]
Sympy is a great work and recently we use it in our [**sympy_to_func project**](https://github.com/PaddlePaddle/community/blob/master/pfcc/paddle-code-reading/sympy_to_func/sympy_to_func.md) for guiding [paddlepaddle](https://www.paddlepaddle.org.cn/en)(a deep learning framework) how to compute a sympy expression when it given. When process a sympy expression, we will first travel in post-order along the corresponding expression tree, and mapping most of the tree node to `paddle.nn.Layer`(you can regard it as `torch.nn.Module` if ur not familiar with paddle) for run-time computing. But we all know that there is no divide(x/y) and substract(x-y) in sympy, using `mul(x,pow(y,-1))` and `x+(-y)` instead, so we inevitably use `x * paddle.pow(y, -1))` API when converting those division node, which may be slower than `x/y` cuz the different behaviour in pow and divide, more obvious in high-order differential case. 

So is there any way to keep division node unchanged in expression tree, such as code blow:

``` py
>>> x, y = sympy.symbols("x y")
>>> z = x / y
>>> z.func
sympy.Div
>>> z.args
(x, y)
```

Looking forward to receiving a reply.

**Comment by oscarbenjamin** [comment]
SymPy does not have any `Div` expression type.

**Reply by HydrogenSulfate** [reply]
Thanks for reply, I will check other ways to optimize.
